geometric_kernels/sampling/samplers.py

```diff
@@ -0,0 +1,41 @@
+"""
+Samplers.
+"""
+from functools import partial
+from typing import Any, Callable, Optional, Tuple
+
+import lab as B
+
+from geometric_kernels.types import FeatureMap
+
+
+def sample_path(feature_map, s, X: B.Numeric, key=None) -> Tuple[Any, Any]:
+    """
+    Given a `feature_map`, compute `s` samples at `X` defined by random state `key`.
+    """
+    features = feature_map(X)  # [N, M]
+
+    key = key or B.global_random_state(B.dtype(X))
+
+    num_features = B.shape(features)[-1]
+
+    key, random_weights = B.randn(key, B.dtype(X), num_features, s)  # [M, S]
+
+    random_sample = B.matmul(features, random_weights)  # [N, S]
+
+    return key, random_sample
+
+
+def sampler(
```

Comment:
Maybe rename this to `sample` or `sample_path`?

Reply:
still stands

Reply:
I think it is worth distinguishing `sampler` and  `sample_path`. The name `sampler` acts as a hint that this is something that returns `sample_path`s.